financial-connections/src/main/java/com/stripe/android/financialconnections/image/StripeImage.kt

```diff
@@ -0,0 +1,84 @@
+package com.stripe.android.financialconnections.image
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.BoxWithConstraintsScope
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.Dp.Companion.Infinity
+import androidx.compose.ui.unit.IntSize.Companion.Zero
+import com.stripe.android.financialconnections.ui.LocalImageLoader
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+@Composable
+internal fun StripeImage(
+    url: String,
+    placeHolder: Painter,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null
+) {
+    val imageLoader = LocalImageLoader.current
+    BoxWithConstraints(modifier) {
+        val (width, height) = calculateBoxSize()
+        val painter: MutableState<Painter> =
+            remember { mutableStateOf(BitmapPainter(ImageBitmap(width, height))) }
+        var loadImageJob: Job?
+        val scope = rememberCoroutineScope()
+
+        DisposableEffect(url) {
+            loadImageJob = scope.launch(Dispatchers.Default) {
+                painter.value = imageLoader.load(url, width, height)
+                    .fold(
+                        onSuccess = { bitmap -> BitmapPainter(bitmap.asImageBitmap()) },
+                        onFailure = { placeHolder }
+                    )
+            }
+            onDispose {
+                painter.value = BitmapPainter(ImageBitmap(width, height))
```

Comment:
Whey do we reset painter value on dispose?

Reply:
The idea here is to free resources when the composable is disposed, ensuring we're not retaining the bitmap on these cases.